core/trino-main/src/main/java/io/trino/execution/buffer/PagesSerdeFactory.java

```diff
@@ -26,21 +32,45 @@ public class PagesSerdeFactory
     private static final int SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES = 64 * 1024;
 
     private final BlockEncodingSerde blockEncodingSerde;
-    private final boolean compressionEnabled;
+    private final CompressionCodec compressionCodec;
 
-    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled)
+    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, CompressionCodec compressionCodec)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
-        this.compressionEnabled = compressionEnabled;
+        this.compressionCodec = requireNonNull(compressionCodec, "compressionCodec is null");
     }
 
     public PageSerializer createSerializer(Optional<SecretKey> encryptionKey)
     {
-        return new PageSerializer(blockEncodingSerde, compressionEnabled, encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+        return new PageSerializer(blockEncodingSerde, createCompressor(compressionCodec), encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
     }
 
     public PageDeserializer createDeserializer(Optional<SecretKey> encryptionKey)
     {
-        return new PageDeserializer(blockEncodingSerde, compressionEnabled, encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+        return new PageDeserializer(blockEncodingSerde, createDecompressor(compressionCodec), encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+    }
+
+    public static Optional<Compressor> createCompressor(CompressionCodec compressionCodec)
+    {
+        return Optional.ofNullable(switch (compressionCodec) {
+            case NONE:
+                yield null;
+            case LZ4:
+                yield new Lz4Compressor();
+            case ZSTD:
+                yield new ZstdCompressor();
+        });
```

Comment:
it'd be more readable to return from switch:


```suggestion
        switch (compressionCodec) {
            case NONE:
                return Optional.empty();
            case LZ4:
                return Optional.of(new Lz4Compressor());
            case ZSTD:
                return Optional.of(new ZstdCompressor());
        };
```